src/app/public/modules/datepicker/datepicker-calendar-inner.component.ts

```diff
@@ -104,21 +123,39 @@ export class SkyDatepickerCalendarInnerComponent implements OnInit, OnChanges {
     40: 'down'
   };
 
+  private customDates: SkyDatepickerCustomDate[] = [];
```

Comment:
☝️ We were able to move away from caching `customDates` in the service, but because of how this component is built, we still need to have the custom dates available to handle events the user triggers like key navigation. For example, the calendar will load, wait for custom dates and then apply them, user uses keyboard arrows to navigate then hits "enter" which calls on the `onKeydown()` method below. This method needs to know if the currently "active" day is disabled to properly handle the key stroke.
To solve this, I am storing the last `customDates` from the consumer. It gets refreshed every time the consumer pushes a new set of data.